crates/vm/levm/src/precompiles.rs

```diff
@@ -232,12 +241,106 @@ pub fn ripemd_160(
     Ok(Bytes::from(output))
 }
 
-fn modexp(
-    _calldata: &Bytes,
-    _gas_for_call: U256,
-    _consumed_gas: &mut U256,
+/// Returns the result of the modexp operation given the input parameters
+pub fn modexp(
+    calldata: &Bytes,
+    gas_for_call: U256,
+    consumed_gas: &mut U256,
 ) -> Result<Bytes, VMError> {
-    Ok(Bytes::new())
+    // If calldata does not reach the required length, we should fill the rest with zeros
+    let calldata = fill_with_zeros(calldata, 96)?;
+
+    let b_size: U256 = calldata
+        .get(0..32)
+        .ok_or(PrecompileError::ParsingInputError)?
+        .into();
+
+    let e_size: U256 = calldata
+        .get(32..64)
+        .ok_or(PrecompileError::ParsingInputError)?
+        .into();
+
+    let m_size: U256 = calldata
+        .get(64..96)
+        .ok_or(PrecompileError::ParsingInputError)?
+        .into();
+
+    if b_size == U256::zero() && m_size == U256::zero() {
+        *consumed_gas = consumed_gas
+            .checked_add(U256::from(MODEXP_STATIC_COST))
+            .ok_or(OutOfGasError::ConsumedGasOverflow)?;
+        return Ok(Bytes::new());
+    }
+
+    // Because on some cases conversions exploded before the if above
+    let b_size = usize::try_from(b_size).map_err(|_| PrecompileError::ParsingInputError)?;
+    let e_size = usize::try_from(e_size).map_err(|_| PrecompileError::ParsingInputError)?;
+    let m_size = usize::try_from(m_size).map_err(|_| PrecompileError::ParsingInputError)?;
+
+    let base_limit = b_size
+        .checked_add(96)
+        .ok_or(InternalError::ArithmeticOperationOverflow)?;
+
+    let exponent_limit = e_size
+        .checked_add(base_limit)
+        .ok_or(InternalError::ArithmeticOperationOverflow)?;
+
+    // The reason I use unwrap_or_default is to cover the case where calldata does not reach the required
+    // length, so then we should fill the rest with zeros. The same is done in modulus parsing
+    let b = calldata.get(96..base_limit).unwrap_or_default();
```

Comment:
To avoid using this I think it's best to fill with zeros up to `96 + Bsize + Esize + Msize` and then we can get the calldata without errors. Imagine that if you have for example an incomplete `e` and it can't get up to the exponent limit then it is going to turn `e` into 0, when it is not.

Reply:
Done in [use get_slice_or_default function](https://github.com/lambdaclass/ethrex/pull/1522/commits/5d0994295decbef922abb6f31e0361bca418d0ee)